utils.py

```diff
@@ -15,6 +16,12 @@ def of_slicer(remaining_data):
     pkts = []
     while data_len > 3:
         length_field = struct.unpack('!H', remaining_data[2:4])[0]
+        ofver = remaining_data[0]
+        # sanity checks: badly formatted packet
+        if ofver not in settings.OPENFLOW_VERSIONS or length_field == 0:
```

Comment:
You are right, @viniarck. The OpenFlow negotiation process is based on the switch sending its highest supported version and the controller sending its version. If the switch supports up to OF 1.5, the sanity check above would lead to a never establishing OF control channel. I've addressed this issue on commit 8654abe. Please check if you see any problems on that solution.